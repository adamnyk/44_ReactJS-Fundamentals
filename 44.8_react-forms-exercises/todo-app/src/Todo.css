.Todo-text {
    display: inline;
}
.Todo button {
    padding: 1px;
    margin: 0 10px;
}