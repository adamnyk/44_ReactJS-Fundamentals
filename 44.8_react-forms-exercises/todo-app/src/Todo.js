import "./Todo.css";
import { IconButton } from '@mui/material';
import DeleteIcon from '@mui/icons-material/Delete';



const Todo = ({ text, id, remove }) => {
	return (
		<div className="Todo">
			<div className="Todo-text">{text}</div>
			
			<IconButton
				aria-label="delete"
				onClick={() => remove(id)}
				data-testid="delete-button"
			>
				<DeleteIcon />
			</IconButton>
		</div>
	);  
};

export default Todo;
