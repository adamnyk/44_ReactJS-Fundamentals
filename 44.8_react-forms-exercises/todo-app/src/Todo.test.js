import { render } from '@testing-library/react';
import Todo from './Todo';

it('Todo renders', () => {
    render(<Todo />);
  });
  
it("matches snapshot", function () {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
  });