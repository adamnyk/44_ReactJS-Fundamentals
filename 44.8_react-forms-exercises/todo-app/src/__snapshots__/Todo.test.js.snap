// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <div
      class="Todo-text"
    />
    <button
      aria-label="delete"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      data-testid="delete-btn"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DeleteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
